src/driver/op.rs

```diff
@@ -183,6 +191,76 @@ where
     }
 }
 
+       
+impl<T> Future for Op<T, MultiCQEFuture>
+where
+    T: Unpin + 'static + Completable + Updateable,
+{
+    type Output = T::Output;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        use std::mem;
+
+        let me = &mut *self;
+
+        CONTEXT.with(|runtime_context| {
+            runtime_context.with_driver_mut(|driver| {
+                let (lifecycle, completions) = driver
+                    .ops
+                    .get_mut(me.index)
+                    .expect("invalid internal state");
+
+                match mem::replace(lifecycle, Lifecycle::Submitted) {
+                    Lifecycle::Submitted => {
+                        *lifecycle = Lifecycle::Waiting(cx.waker().clone());
+                        Poll::Pending
+                    }
+                    Lifecycle::Waiting(waker) if !waker.will_wake(cx.waker()) => {
+                        *lifecycle = Lifecycle::Waiting(cx.waker().clone());
+                        Poll::Pending
+                    }
+                    Lifecycle::Waiting(waker) => {
+                        *lifecycle = Lifecycle::Waiting(waker);
+                        Poll::Pending
+                    }
+                    Lifecycle::Ignored(..) => unreachable!(),
+                    Lifecycle::Completed(cqe) => {
+                        // This is possible. We may have previously polled a CompletionList,
+                        // and the final CQE registered as Completed
+                        driver.ops.remove(me.index);
+                        me.index = usize::MAX;
+                        Poll::Ready(me.data.take().unwrap().complete(cqe))
+                    }
+                    Lifecycle::CompletionList(indices) => {
+                        let mut data = me.data.take().unwrap();
+
+                        // Traverse the CqeResult list, calling update on the Op on all Cqe's flagged `more`
+                        // If the final Cqe is present, return it in an Option
```

Comment:
Rather than 'Traverse', could this say 'Consume' the completion list?

And the second line, something like

// If the final Cqe is present, cleanup and return Poll::Ready?

Or is that too much on the nose?